packages/blade/src/components/Icons/RupeeIcon/RupeeIcon.tsx

```diff
@@ -1,9 +1,8 @@
-import type { ReactElement } from 'react';
 import { Svg, Path } from '../_Svg';
-import type { IconProps } from '..';
+import type { IconComponent } from '..';
 import useIconProps from '../useIconProps';
 
-const RupeeIcon = ({ size, color }: IconProps): ReactElement => {
+const RupeeIcon: IconComponent = ({ size, color }) => {
```

Comment:
what is the difference between writing it this way vs that way?

I understand with this we are typing the variable and in the earlier approach we were typing the object arg as well as the function arg, but when to use which approach?

cc: @anuraghazra

Reply:
Depends, One is a function declaration, where you are typing the function when defining it's implementation. 

And the second one is a function type signature or function call signature (whatever you wanna call it)

Here's the difference: https://tsplay.dev/wEPRvW

Reply:
In my non-ts expert understanding - Its now easier for our consumers to create a new component by using `IconComponent` type directly instead of specifying the props & return type

This also allows us to use `IconComponent` in places like Button where want to accept a `icon` prop and we can make it's type as `icon: IconComponent` so that our expectations are clearly mentioned